call_option.go

```diff
@@ -37,27 +37,55 @@ import (
 	"google.golang.org/grpc/codes"
 )
 
+// CallOption is an option used by Invoke
 type CallOption interface {
 	Resolve(*CallSettings)
 }
 
-type retryFuncOption func() RetryFunc
+// Retryer is used by Invoke to determine retry behavior.
+type Retryer interface {
+	// Retry reports whether a request should be retried and how long to pause before retrying
+	// if the previous attempt returned with err.
+	// Invoke never calls ShouldRetry with nil error.
+	Retry(err error) (pause time.Duration, shouldRetry bool)
+}
+
+type retryerOption func() Retryer
 
-func (o retryFuncOption) Resolve(s *CallSettings) {
+func (o retryerOption) Resolve(s *CallSettings) {
 	s.Retry = o
 }
 
-func WithRetry(fn func() RetryFunc) CallOption {
-	return retryFuncOption(fn)
+// WithRetry sets CallSettings.Retry to fn.
+func WithRetry(fn func() Retryer) CallOption {
+	return retryerOption(fn)
 }
 
+// Backoff implements Retryer, retrying on error codes in Codes,
+// and performing exponential backoffs between retries.
+//
+// A request is retried if and only if the error is a GRPC error
+// whose error code is stored in Codes.
+// The wait time between retries is a random value between 0 and the "retry envelope".
+// The envelope starts at Initial and increases by the factor of Multiplier every retry,
+// but is capped at Max.
 type Backoff struct {
+	// Initial is the initial value of the retry envelope, defaults to 1 second.
 	Initial time.Duration
-	Max     time.Duration
-	Mult    float64
-	Codes   []codes.Code
+
+	// Max is the maximum value of the retry envelope, defaults to 30 seconds.
+	Max time.Duration
+
+	// Multiplier is the factor by which the retry envelope increases.
+	// It should be greater than 1 and defaults to 2.
+	Multiplier float64
+
+	// Codes are the GRPC error codes that Backoff should retry.
+	Codes []codes.Code
```

Comment:
Or maybe:

WithRetry(func () Retry { return OnCodes(myRetryCodes, Backoff{Initial: time.Second}) })

Less magic, and reads better.

@broady care to bikeshed?